### PR TITLE
Memoize Shopper Insights API Call

### PR DESCRIPTION
**Pull request by ravishekhar**
### Description
* Refactor the existing shopper insights tests to mock the request method of external belter module instead of internal callRestAPI
* Memoize the Shopper Insights API Call to avoid duplicate calls with exact same set of parameters. 

### Groups who should review (if applicable)

❤️ Thank you!

### CODE REVIEW

File: src/api/api.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* @flow */
 
 import { getPartnerAttributionID, getSessionID } from "@paypal/sdk-client/src";
-import { request } from "@krakenjs/belter/src";
+import { inlineMemoize, request } from "@krakenjs/belter/src";
```
Comment (jshawl) [inline src/api/api.js:4]: what is the difference between [`inlineMemoize`](https://github.com/krakenjs/belter/blob/d0c0868c996aea92d1ad9a6c8662da7a2c200b5b/src/util.js#L334) and [`memoize`](https://github.com/krakenjs/belter/blob/d0c0868c996aea92d1ad9a6c8662da7a2c200b5b/src/util.js#L188)

just curious..

Comment (jshawl) [reply src/api/api.js:4]: oh reading below it seems like it will cache based on the args passed

File: src/api/shopper-insights/component.test.js [hunk 2]
```diff
@@ -67,20 +65,87 @@ describe("shopper insights component - getRecommendedPaymentMethods()", () => {
...
+    expect(response1).toEqual({
+      isPayPalRecommended: true,
+      isVenmoRecommended: false,
+    });
+    expect(response2).toEqual({
+      isPayPalRecommended: true,
+      isVenmoRecommended: false,
+    });
+    expect.assertions(6);
+  });
```
Comment (jshawl) [inline src/api/shopper-insights/component.test.js:106]: awesome test coverage! The way you structured this test made it super easy to understand why the change is being made 🏅

File: src/api/shopper-insights/component.test.js [hunk 3]
```diff
@@ -92,35 +157,39 @@ describe("shopper insights component - getRecommendedPaymentMethods()", () => {
...
         phone: {
           countryCode: "1",
-          nationalNumber: "2345678901",
+          nationalNumber: "2345678906",
         },
       },
     });
 
-    expect(callRestAPI).toHaveBeenCalledWith(
+    expect(request).toHaveBeenCalledWith(
```
Comment (jshawl) [inline src/api/shopper-insights/component.test.js:185]: is there a reason you chose to expect `request` to be called instead of expecting `callMemoizedRestAPI` to be called? just curious (the change from `data` to `json` prompted this question for me)

**Review by jshawl — APPROVED** [review]
LGTM I had a few non-blocking questions just for learning purposes

**Review by wsbrunson — APPROVED** [review]

File: src/api/api.js (hunk 1 above)
Comment (ravishekhar) [reply src/api/api.js:4]: Great question. You are 💯 right about args. Although memoize also creates cacheKey based on method arguments, but inlineMemoize generally provides more control on the cache key e.g. we can serialize complex objects as per the needs. 

Also with inlineMemoize the cache lives with method but memoize will end up creating a closure for the cache.
